examples/02_model_collaborative_filtering/sar_deep_dive.ipynb

```diff
@@ -576,10 +489,10 @@
      "text": [
       "Model:\n",
       "Top K:\t\t 10\n",
-      "MAP:\t\t 0.095544\n",
-      "NDCG:\t\t 0.350232\n",
-      "Precision@K:\t 0.305726\n",
-      "Recall@K:\t 0.164690\n"
+      "MAP:\t\t 0.113796\n",
+      "NDCG:\t\t 0.384809\n",
+      "Precision@K:\t 0.331707\n",
+      "Recall@K:\t 0.182571\n"
```

Comment:
@anargyri I run the notebook with python 3.6-3.9 and I always got the same results. So I believe that the numbers are correct.

An hypothesis of what could have happened is that we haven't updated this notebook since before 2020 https://github.com/microsoft/recommenders/commits/main/examples/02_model_collaborative_filtering/sar_deep_dive.ipynb and in the mean time, there was been some changes, like the TOP_K issue Chuyang found https://github.com/microsoft/recommenders/commit/68b60c04819327efb54805d1215ef125b269a919. That could explain the different numbers. Another option is the splitter, again, since it's been a while that we haven't touched the notebook, we might have not detected them.

Reply:
Yes, I think you are right. Some time in the past there was a change but the notebook outputs were not checked in properly. Btw I think we do not ensure against something like this in the unit tests, do we? I.e. we just do a test that this notebook executes without error but we don't test that the results don't change significantly (I remember that this can be a source of many failing tests, but maybe we should revisit?)

Reply:
Here the discrepancy is around 20% which should raise a flag imo. I mean, the error tolerance can be high enough, but at least there should be some error checking, whereas now there is none.